src/wallet/userKeys.ts

```diff
@@ -36,9 +37,15 @@ export class UserSecretKey {
         return new UserPublicKey(buffer);
     }
 
-    sign(message: Buffer | Uint8Array): Buffer {
+    static generate(): UserSecretKey {
+        const keyPair = nacl.sign.keyPair(); // Generates a new signing keypair
+        const secretKey = keyPair.secretKey.subarray(0, 32); // Extract only the private key part
+        return new UserSecretKey(secretKey);
+    }
+
+    sign(message: Buffer | Uint8Array): Uint8Array {
```

Comment:
I think we can have `sign(message: Uint8Array): Uint8Array` since `Buffer` is an implementation of `Uint8Array`.